python/langsmith/client.py

```diff
@@ -145,16 +116,62 @@ def _serialize_json(obj: Any) -> str:
         return str(obj)
 
 
-def close_session(session: requests.Session) -> None:
+def close_client(client: httpx.Client) -> None:
     """Close the session.
 
     Parameters
     ----------
-    session : Session
-        The session to close.
+    client : client
+        The httpx client to close.
     """
-    logger.debug("Closing Client.session")
-    session.close()
+    if client.is_closed:
+        return
+    logger.debug("Closing Client._client")
+    client.close()
+
+
+def close_async_client(client: httpx.AsyncClient) -> None:
+    logger.debug("Closing Client._aclient")
+    if client.is_closed:
+        return
+    coro = client.aclose()
+    try:
+        # Raises RuntimeError if there is no current event loop.
+        asyncio.get_running_loop()
+        loop_running = True
+    except RuntimeError:
+        loop_running = False
+
+    if loop_running:
+        # If we try to submit this coroutine to the running loop
+        # we end up in a deadlock, as we'd have gotten here from a
+        # running coroutine, which we cannot interrupt to run this one.
+        # The solution is to create a new loop in a new thread.
+        with concurrent.futures.ThreadPoolExecutor(1) as executor:
+            executor.submit(_run_coro, coro).result()
+    else:
+        _run_coro(coro)
+
+
+def _run_coro(coro: Coroutine) -> None:
+    if hasattr(asyncio, "Runner"):
+        # Python 3.11+
+        # Run the coroutines in a new event loop, taking care to
+        # - install signal handlers
+        # - run pending tasks scheduled by `coros`
+        # - close asyncgens and executors
+        # - close the loop
+        with asyncio.Runner() as runner:
+            # Run the coroutine, get the result
+            runner.run(coro)
```

Comment:
This line still giving a `RuntimeError: Event loop is closed`

it was locking previously with simple stuff.